### DISCUSSION TITLE
Shared folder between Windows and Linux

### DISCUSSION THREAD

**Discussion by Makhaira** [abraunegg/onedrive#1961] [Q&A] [answered]
Ok, system in question is dual booting Linux Mint with Windows 11. There is a separate Data partition in NTFS.

In Windows, my onedrive client keeps a full offline sync of all my files in the Data drive.

In Linux, I intend to use this client to provide the sync/offline copy.

Now, my question is, is there any way to avoid duplication? Linux is very unhappy about any of the offline files at the moment due to referrals that it doesn't like (I understand this is typical of OneDrive on an NTFS drive) so I can't access it from within Linux to point the OneDrive for Linux client at. Is this something that changes by using this client, or am I always going to need two copies of my onedrive folder on the machine for each OS to access?

Thanks.

**Comment by abraunegg** [comment, marked as answer]
@Makhaira 
> In Windows, my onedrive client keeps a full offline sync of all my files in the Data drive.

Are you 100% sure of this, as, the Windows OneDrive Client typically uses 'files on-demand' by default - meaning that the files are not really present as full files on your local disk. To fix this, you need to read https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md#configuring-the-client-for-use-in-dual-boot-windows--linux-situations

> Now, my question is, is there any way to avoid duplication? Linux is very unhappy about any of the offline files at the moment due to referrals that it doesn't like (I understand this is typical of OneDrive on an NTFS drive) so I can't access it from within Linux to point the OneDrive for Linux client at.

Correct - see point above.

> Is this something that changes by using this client, or am I always going to need two copies of my onedrive folder on the machine for each OS to access?

You can have just one copy of your OneDrive data ... however the advice here is:
1. Ensure that you configure the Windows client correctly - see point above
2. Update your Windows configuration so that your OneDrive data is stored on a non-NTFS partition - an exFAT formatted partition will give you the most compatibility as read|write of NTFS with all ACLs|security flags|other attributes could cause other issues under Linux
3. Mount this exFAT partition under Linux in your desired location
4. Symbolically link your users OneDrive folder to this location

By doing this you will achieve:
* One copy of data - full data contents
* Data is accessible and has maximum compatibility under Linux and Windows

**Reply by Makhaira** [reply]
Thanks for the speedy response!

It definitely keeps a full offline set as I've configured it to do this - I'm often in areas without wifi coverage so it's important to me.

I didn't know about exFat as a viable option, I'll try that. Thank you very much for the help!

**Comment by Makhaira** [comment]
Annoyingly OneDrive no longer lets you set it up on an exFAT drive, it says it must be an NTFS drive! Talk about unhelpful software....
